tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -214,12 +220,188 @@ async def with_ci_connector_ops(context: PipelineContext) -> Container:
     return await with_installed_python_package(context, python_with_git, CI_CONNECTOR_OPS_SOURCE_PATH, exclude=["pipelines"])
 
 
+def with_dockerd_service(
+    context: ConnectorTestContext, shared_volume: Optional(Tuple[str, CacheVolume]) = None, docker_cache_volume_name: Optional[str] = None
+) -> Container:
+    """Create a container running dockerd, exposing its 2375 port, can be used as the docker host for docker-in-docker use cases.
+
+    Args:
+        context (ConnectorTestContext): The current connector test context.
+        shared_volume (Optional, optional): A tuple in the form of (mounted path, cache volume) that will be mounted to the dockerd container. Defaults to None.
+        docker_cache_volume_name (Optional[str], optional): The name of the docker cache volume. Changing the volume name provides docker context isolation. Defaults to None.
+
+    Returns:
+        Container: The container running dockerd as a service.
+    """
+    dind = (
+        context.dagger_client.container()
+        .from_("docker:23.0.1-dind")
+        .with_mounted_cache(
+            "/var/lib/docker",
+            context.dagger_client.cache_volume(f"{slugify(context.connector.technical_name)}-docker-lib"),
```

Comment:
@cpdeethree do you need to explicitly declare a new volume to persist to S3 on the infra side to make these volume shareable across builds?